### PR TITLE
cockroach-oss: delete build target

### PR DESCRIPTION
**Pull request by rickystewart**
With the new license, this target is misleading and confers no advantage and does not have a different license than the normal `cockroach` target.

Epic: DEVINF-1246
Release note: None

### CODE REVIEW

File: pkg/cmd/cockroach-oss/main.go [hunk 1]
```diff
@@ -1,17 +0,0 @@
...
-// Use of this software is governed by the CockroachDB Software License
-// included in the /LICENSE file.
-
-// cockroach-oss is an entry point for a CockroachDB binary that excludes all
-// CCL-licensed code.
-package main
-
-import (
-	"github.com/cockroachdb/cockroach/pkg/cli"
-	_ "github.com/cockroachdb/cockroach/pkg/ui/distoss" // web UI init hooks
```
Comment (jlinder) [inline pkg/cmd/cockroach-oss/main.go:12]: We should also remove this package.

**Review by jlinder — CHANGES_REQUESTED** [review]
A few other updates to make:

- `cockroach-oss` is also referenced in these files: `.github/CODEOWNERS`, `build/GNUmakefile.obsolete`

**Comment by rickystewart** [comment]
> cockroach-oss is also referenced in these files: .github/CODEOWNERS, build/GNUmakefile.obsolete

I'll update `.github/CODEOWNERS`.

We don't really update `build/GNUmakefile.obsolete` -- it's already stale and doesn't get changes as we make changes to `bazel` build stuff. In a separate PR we'll delete that file (as discussed on Slack)

**Review by jlinder — APPROVED** [review]

**Comment by rickystewart** [comment]
bors r=jlinder

**Comment by jlinder** [comment]
@rickystewart please also backport this to release-23.1. It's still in LTS.

**Pull request merged**